nixos/modules/system/activation/activation-script.nix

```diff
@@ -158,8 +158,11 @@ in
         ${pkgs.utillinux}/bin/mount -o "remount,size=${config.boot.devSize}" none /dev
         ${pkgs.utillinux}/bin/mount -o "remount,size=${config.boot.devShmSize}" none /dev/shm
         ${pkgs.utillinux}/bin/mount -o "remount,size=${config.boot.runSize}" none /run
+      '' + optionalString config.boot.tmpOnTmpfs ''
+        if [ mountpoint -q /tmp ]; then
```

Comment:
That said... the stage-2-init.sh script doesn't seem to do much error handling of mount points fail during boot?